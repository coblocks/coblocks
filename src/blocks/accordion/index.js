/**
 * Styles.
 */
import './styles/editor.scss';
import './styles/style.scss';

/**
 * Internal dependencies
 */
import edit from './edit';
import icon from './icon';
import metadata from './block.json';
import transforms from './transforms';

/**
 * WordPress dependencies
 */
import { __, _x } from '@wordpress/i18n';
import { InnerBlocks } from '@wordpress/block-editor';

/**
 * Block constants
 */
const { name, category, attributes } = metadata;

const settings = {
	title: _x( 'Accordion', 'block title', 'coblocks' ),
	description: __( 'Organize content within collapsable accordion items.', 'coblocks' ),
	icon,
	keywords: [ _x( 'tabs', 'block keyword', 'coblocks' ), _x( 'faq', 'block keyword', 'coblocks' ), 'coblocks' ],
	supports: {
		align: [ 'wide', 'full' ],
		html: false,
	},
	attributes,
	transforms,
	edit,
	save() {
		return (
			<div>
				<InnerBlocks.Content />
			</div>
		);
	},
};

export { name, category, metadata, settings };
