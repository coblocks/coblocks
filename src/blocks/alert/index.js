/**
 * Styles.
 */
import './styles/editor.scss';
import './styles/style.scss';

/**
 * Internal dependencies
 */
import deprecated from './deprecated';
import edit from './edit';
import icon from './icon';
import metadata from './block.json';
import save from './save';
import transforms from './transforms';

/**
 * WordPress dependencies
 */
import { __, _x } from '@wordpress/i18n';

/**
 * Block constants
 */
const { name, category, attributes } = metadata;

const settings = {
	title: _x( 'Alert', 'block name', 'coblocks' ),
	description: __( 'Provide contextual feedback messages or notices.', 'coblocks' ),
	icon,
	keywords: [ _x( 'notice', 'block keyword', 'coblocks' ), _x( 'message', 'block keyword', 'coblocks' ), 'coblocks' ],
	styles: [
		{ name: 'info', label: _x( 'Info', 'block style', 'coblocks' ), isDefault: true },
		{ name: 'success', label: _x( 'Success', 'block style', 'coblocks' ) },
		{ name: 'warning', label: _x( 'Warning', 'block style', 'coblocks' ) },
		{ name: 'error', label: _x( 'Error', 'block style', 'coblocks' ) },
	],
	supports: {
		align: true,
		alignWide: false,
		alignFull: false,
	},
	example: {
		attributes: {
			title: __( 'This is an alert block', 'coblocks' ),
			value: __( 'An alert is a message that displays outside the flow of typical content. Alerts provide contextual feedback, typically asking readers to take an action.', 'coblocks' ),
		},
	},
	attributes,
	transforms,
	edit,
	save,
	deprecated,
};

export { name, category, metadata, settings, attributes };
