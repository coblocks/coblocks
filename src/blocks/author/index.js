/**
 * Styles.
 */
import './styles/style.scss';
import './styles/editor.scss';

/**
 * Internal dependencies
 */
import deprecated from './deprecated';
import edit from './edit';
import icon from './icon';
import metadata from './block.json';
import save from './save';
import transforms from './transforms';

/**
 * WordPress dependencies
 */
import { __, _x } from '@wordpress/i18n';

/**
 * Block constants
 */
const { name, category, attributes } = metadata;

const settings = {
	title: _x( 'Author', 'block name', 'coblocks' ),
	description: __( 'Add an author biography to build credibility and authority.', 'coblocks' ),
	icon,
	keywords: [ _x( 'biography', 'block keyword', 'coblocks' ), _x( 'profile', 'block keyword', 'coblocks' ), 'coblocks' ],
	example: {
		attributes: {
			name: 'Jane Doe',
			biography: __( 'Born to express, not to impress. A maker making the world I want.', 'coblocks' ),
			imgUrl: '/wp-content/plugins/coblocks/dist/images/examples/author.jpg',
		},
	},
	attributes,
	transforms,
	edit,
	save,
	deprecated,
};

export { name, category, metadata, settings };
