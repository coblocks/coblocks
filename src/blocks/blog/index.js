/**
 * Internal dependencies
 */
import edit from './edit';
import icons from './icons';

/**
 * WordPress dependencies
 */
const { __, _x } = wp.i18n;

const name = 'blog';

const title = __( 'Blog' );

const icon = icons.blog;

const settings = {
	title: title,
	description: __( 'Display a list of your most recent posts.' ),
	icon,
	category: 'widgets',
	keywords: [ __( 'blog posts' ) ],
	styles: [
		{ name: 'grid', label: _x( 'Grid', 'block style' ), isDefault: true },
		{ name: 'list', label: _x( 'List', 'block style' ) },
		{ name: 'carousel', label: _x( 'Carousel', 'block style' ) },
	],
	supports: {
		align: [ 'wide', 'full' ],
		html: false,
	},
	edit,
};

export { name, title, icon, settings };
