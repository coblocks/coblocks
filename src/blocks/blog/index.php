<?php
/**
 * Server-side rendering of the `blog` block.
 *
 * @package WordPress
 */

/**
 * Renders the `blog` block on server.
 *
 * @param array $attributes The block attributes.
 *
 * @return string Returns the post content with latest posts added.
 */
function render_block_blog( $attributes ) {
	$args = array(
		'posts_per_page'   => $attributes['postsToShow'],
		'post_status'      => 'publish',
		'order'            => $attributes['order'],
		'orderby'          => $attributes['orderBy'],
		'suppress_filters' => false,
	);

	if ( isset( $attributes['categories'] ) ) {
		$args['category'] = $attributes['categories'];
	}

	if ($attributes['postFeedType'] === 'external' && $attributes['externalRssUrl']) {

		$recent_posts = fetch_feed($attributes['externalRssUrl']);

		if ( is_wp_error( $recent_posts ) ) {
			return '<div class="components-placeholder"><div class="notice notice-error"><strong>' . __( 'RSS Error:' ) . '</strong> ' . $recent_posts->get_error_message() . '</div></div>';
		}

		if ( ! $recent_posts->get_item_quantity() ) {
			// PHP 5.2 compatibility. See: http://simplepie.org/wiki/faq/i_m_getting_memory_leaks.
			$recent_posts->__destruct();
			unset( $recent_posts );
			return '<div class="components-placeholder"><div class="notice notice-error">' . __( 'An error has occurred, which probably means the feed is down. Try again later.' ) . '</div></div>';
		}

		$recent_posts  = $recent_posts->get_items( 0, $attributes['postsToShow'] );
		$formattedPosts = extract_external_info($recent_posts);
	} else {
		$recent_posts = get_posts( $args );
		$formattedPosts = extract_internal_info($recent_posts);
	}

	$blogLayout = null;

	if ( isset( $attributes['className'] ) && strpos( $attributes['className'], 'is-style-list' ) !== false ) {
		$blogLayout = 'list';
	} elseif (isset( $attributes['className'] ) && strpos( $attributes['className'], 'is-style-grid' ) !== false) {
		$blogLayout = 'grid';
	} else {
		$blogLayout = 'carousel';
	}

	if ($blogLayout === 'carousel') {
		return build_carousel_block_content($formattedPosts, $attributes);
	} else {
		return build_non_carousel_block_content($formattedPosts, $attributes);
	}
}

function build_carousel_block_content($posts, $attributes) {

	$arrows = $attributes['prevNextButtons'] ? 'true' : 'false';
	$auto_play = $attributes['autoPlay'] ? 'true' : 'false';
	$draggable = (string) $attributes['draggable'] ? 'true' : 'false';

	$class = 'wp-block-coblocks-blog';

	if ($attributes['align'] !== '') {
		$class .= ' align'.$attributes['align'];
	}

	if ( isset( $attributes['className'] ) ) {
		$class .= ' ' . $attributes['className'];
	}

	$block_content = <<<EOL
<div class="carousel-container $class" 
data-slick='{"slidesToShow": {$attributes['visibleItems']},
"slidesToScroll": 1,
"arrows": $arrows,
"autoPlay": $auto_play,
"autoPlaySpeed": {$attributes['autoPlaySpeed']},
"infinite": true,
"adaptiveHeight": false,
"draggable": $draggable
}'>
EOL;

	$list_items_markup = '';

	foreach ($posts as $post) {
		$list_items_markup .= '<div class="coblocks-blog-post--item">';
		$list_items_markup .= '<div class="coblocks-blog-post--item-inner">';

		if ( $post['thumbnailURL'] !== null && $post['thumbnailURL']) {
			$list_items_markup .= sprintf(
				'<div class="wp-block-coblocks-blog__post-image-wrapper" style="background-image:url(%2$s)"><a href="%1$s"></a></div>',
				$post['postLink'],
				$post['thumbnailURL']
			);
		}

		$item_info_class = 'wp-block-coblocks-blog__post-info ';
		if ($post['thumbnailURL'] === null || !$post['thumbnailURL']) {
			$item_info_class .= 'full-height ';
		}
		$list_items_markup .= sprintf(
			'<div class="%1$s"</div>',
			$item_info_class
		);

		if ( isset( $attributes['displayPostDate'] ) && $attributes['displayPostDate'] ) {
			$list_items_markup .= sprintf(
				'<time datetime="%1$s" class="wp-block-coblocks-blog__post-date">%2$s</time>',
				$post['date'],
				$post['dateReadable']
			);
		}

		$title = $post['title'];
		if ( ! $title ) {
			$title = __( '(Untitled)' );
		}
		$list_items_markup .= sprintf(
			'<h5 href="%1$s">%2$s</h5>',
			$post['postLink'],
			$title
		);

		if ( isset( $attributes['displayPostContent'] ) && $attributes['displayPostContent']) {
			$postExcerpt = $post['postExcerpt'];
			$trimmed_excerpt = esc_html( wp_trim_words( $postExcerpt, $attributes['excerptLength'], ' &hellip; ' ) );

			$list_items_markup .= sprintf(
				'<div class="wp-block-coblocks-blog__post-excerpt"><p>%1$s</p></div>',
				$trimmed_excerpt
			);
		}

		if ( isset( $attributes['displayPostLink'] ) && $attributes['displayPostLink']) {
			$list_items_markup .= sprintf(
				'<div class="wp-block-coblocks-blog__post-read-more"><a href="%2$s">%1$s</a></div>',
				$attributes['postLink'],
				$post['postLink']
			);
		}

		$list_items_markup .= '</div></div></div>';
	}

	$block_content .= $list_items_markup;
	$block_content .= '</div>';

	return $block_content;
}

function build_non_carousel_block_content($posts, $attributes) {
	$class = 'wp-block-coblocks-blog wp-block-coblocks-blog__list';

	$blogLayout = strpos($attributes['className'], 'is-style-grid') !== false ? 'grid' : 'list';

	if ( isset( $attributes['listPosition'] ) && $blogLayout === 'list' ) {
		$class .= ' image-to-' . $attributes['listPosition'];
	}

	if ( isset( $attributes['columns'] ) && 'grid' === $blogLayout ) {
		$class .= ' columns-' . $attributes['columns'];
	}

	if ( isset( $attributes['displayPostDate'] ) && $attributes['displayPostDate'] ) {
		$class .= ' has-dates';
	}

	if ( isset( $attributes['className'] ) ) {
		$class .= ' ' . $attributes['className'];
	}

	$block_content = <<<EOL
<ul class="$class">
EOL;

	$list_items_markup = '';

	foreach ( $posts as $post ) {

		$list_items_markup .= '<li>';

		if ( $post['thumbnailURL'] !== null &&  $post['thumbnailURL']) {
			$list_items_markup .= sprintf(
				'<div class="wp-block-coblocks-blog__post-image-wrapper" style="background-image:url(%2$s)"><a href="%1$s"></a></div>',
				$post['postLink'],
				$post['thumbnailURL']
			);
		}

		$list_items_markup .= '<div class="wp-block-coblocks-blog__post-info">';

		if ( isset( $attributes['displayPostDate'] ) && $attributes['displayPostDate'] ) {
			$list_items_markup .= sprintf(
				'<time datetime="%1$s" class="wp-block-coblocks-blog__post-date">%2$s</time>',
				$post['date'],
				$post['dateReadable']
			);
		}

		$title = $post['title'];
		if ( ! $title ) {
			$title = __( '(Untitled)' );
		}
		$list_items_markup .= sprintf(
			'<h5><a href="%1$s">%2$s</a></h5>',
			$post['postLink'],
			$title
		);

		if ( isset( $attributes['displayPostContent'] ) && $attributes['displayPostContent']) {
			$postExcerpt = $post['postExcerpt'];
			$trimmed_excerpt = esc_html( wp_trim_words( $postExcerpt, $attributes['excerptLength'], ' &hellip; ' ) );

			$list_items_markup .= sprintf(
				'<div class="wp-block-coblocks-blog__post-excerpt"><p>%1$s</p></div>',
				$trimmed_excerpt
			);
		}

		if ( isset( $attributes['displayPostLink'] ) && $attributes['displayPostLink']) {
			$list_items_markup .= sprintf(
				'<div class="wp-block-coblocks-blog__post-read-more"><a href="%2$s">%1$s</a></div>',
				$attributes['postLink'],
				$post['postLink']
			);
		}


		$list_items_markup .= "</li>";
	}

	$block_content .= $list_items_markup;
	$block_content .= '</ul>';


	return $block_content;
}

function extract_external_info($posts) {
	$formatted_posts = [];

	foreach ($posts as $post) {
		$formatted_post = null;

		$formatted_post['date'] = date_i18n( get_option('c'), $post->get_date( 'U' ) );
		$formatted_post['dateReadable'] = date_i18n( get_option('date_format'), $post->get_date( 'U' ) );
		$formatted_post['title'] = $title = esc_html( trim( strip_tags( $post->get_title() ) ) );
		$formatted_post['postLink'] = esc_url( $post->get_link() );
		$formatted_post['postExcerpt'] = html_entity_decode( $post->get_description(), ENT_QUOTES, get_option( 'blog_charset' ) );

		$output = preg_match_all('/<img.+src=[\'"]([^\'"]+)[\'"].*>/i', $post->get_content(), $matches);

		$first_img = false;
		if ($matches && !empty($matches[1])) {
			$first_img = $matches[1][0];
		}

		$formatted_post['thumbnailURL'] = $first_img;

		$formatted_posts[] = $formatted_post;
	}

	return $formatted_posts;
}

function extract_internal_info($posts) {
	$formatted_posts = [];

	foreach ($posts as $post) {
		$formatted_post = null;

		$formatted_post['thumbnailURL'] = get_the_post_thumbnail_url($post);;
		$formatted_post['date'] = esc_attr( get_the_date( 'c', $post ) );
		$formatted_post['dateReadable'] = esc_html( get_the_date( '', $post ) );
		$formatted_post['title'] = get_the_title( $post );
		$formatted_post['postLink'] = esc_url( get_permalink( $post ) );

		$post_excerpt = $post->post_excerpt;
		if ( ! ( $post_excerpt ) ) {
			$post_excerpt = $post->post_content;
		}

		$formatted_post['postExcerpt'] = $post_excerpt;

		$formatted_posts[] = $formatted_post;
	}

	return $formatted_posts;
}

/**
 * Registers the `blog` block on server.
 */
function register_block_blog() {
	// Return early if this function does not exist.
	if ( ! function_exists( 'register_block_type' ) ) {
		return;
	}

	$dir = CoBlocks()->asset_source( 'js' );

	wp_register_script(
		'coblocks-slick-initializer',
		$dir . 'coblocks-slick-initializer' . COBLOCKS_ASSET_SUFFIX . '.js',
		array( 'jquery')
	);

	register_block_type(
		'coblocks/blog',
		array(
			'attributes'      => array(
				'listPosition'			  => array(
					'type'    => 'string',
					'default' => 'left',
				),
				'align'                   => array(
					'type'    => 'string',
					'default' => 'wide',
				),
				'className'               => array(
					'type' => 'string',
				),
				'categories'              => array(
					'type' => 'string',
				),
				'postsToShow'             => array(
					'type'    => 'number',
					'default' => 5,
				),
				'displayPostContent'      => array(
					'type'    => 'boolean',
					'default' => false,
				),
				'displayPostLink'		  => array(
					'type'    => 'boolean',
					'default' => false,
				),
				'displayCategories'		  => array(
					'type'    => 'boolean',
					'default' => false,
				),
				'postLink'		  		  => array(
					'type'    => 'string',
					'default' => 'Continue Reading',
				),
				'postFeedType'			  => array(
					'type'	  => 'string',
					'default' => 'internal',
				),
				'externalRssUrl'		  => array(
					'type'	  => 'string',
					'default' => ''
				),
				'excerptLength'           => array(
					'type'    => 'number',
					'default' => 55,
				),
				'displayPostDate'         => array(
					'type'    => 'boolean',
					'default' => false,
				),
				'columns'                 => array(
					'type'    => 'number',
					'default' => 2,
				),
				'order'                   => array(
					'type'    => 'string',
					'default' => 'desc',
				),
				'orderBy'                 => array(
					'type'    => 'string',
					'default' => 'date',
				),
				'prevNextButtons'		  => array(
					'type'	  => 'boolean',
					'default' => true
				),
				'draggable'		  => array(
					'type'	  => 'boolean',
					'default' => true
				),
				'autoPlay'		  => array(
					'type'	  => 'boolean',
					'default' => false
				),
				'autoPlaySpeed'	  => array(
					'type'	  => 'string',
					'default' => 3000
				),
				'visibleItems'	  => array(
					'type'	  => 'number',
					'default' => 2
				)
			),
			'render_callback' => 'render_block_blog',
			'editor_script'   => 'coblocks-slick-initializer',
		)
	);
}
add_action( 'init', 'register_block_blog' );
