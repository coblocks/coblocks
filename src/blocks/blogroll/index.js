/**
 * Internal dependencies
 */
import edit from './edit';
import icons from './icons';
import metadata from './block.json';

/**
 * WordPress dependencies
 */
const { __ } = wp.i18n;

/**
 * Block constants
 */
const { name, category, supports } = metadata;

const title = __( 'Blogroll' );

const icon = icons.blog;

const settings = {
	title: title,
	description: __( 'Display a list of your most recent posts.' ),
	icon,
	keywords: [ __( 'posts' ), __( 'articles' ), 'coblocks' ],
	supports,
	edit,
	save: function( props ) {
		return null;
	},
};

export { name, category, title, icon, settings };
