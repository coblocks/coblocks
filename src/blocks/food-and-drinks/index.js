/**
 * Styles.
 */
import './styles/editor.scss';
import './styles/style.scss';

/**
 * Internal dependencies.
 */
import edit from './edit';
import icon from './icon';
import metadata from './block.json';
import save from './save';

/**
 * WordPress dependencies.
 */
import { __, _x } from '@wordpress/i18n';

/**
 * Block constants.
 */
const { name, category, attributes } = metadata;

const settings = {
	title: _x( 'Food & Drinks', 'block name', 'coblocks' ),
	description: __( 'Display a menu or price list.', 'coblocks' ),
	icon,
	keywords: [ _x( 'restaurant', 'block keyword', 'coblocks' ), _x( 'menu', 'block keyword', 'coblocks' ) ],
	supports: {
		align: [ 'wide' ],
	},
	getEditWrapperProps( attributes ) {
		return { 'data-columns': attributes.columns };
	},
	attributes,
	edit,
	save,
};

export { name, category, metadata, settings };
