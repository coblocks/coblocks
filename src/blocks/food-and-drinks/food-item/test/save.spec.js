/**
 * External dependencies
 */
import '@testing-library/jest-dom/extend-expect';
import { registerBlockType, createBlock, serialize } from '@wordpress/blocks';

/**
 * Internal dependencies.
 */
import { name, settings } from '../index';

// Make variables accessible for all tests.
let block;
let serializedBlock;

describe( name, () => {
	beforeAll( () => {
		// Register the block.
		registerBlockType( name, { category: 'common', ...settings } );
	} );

	beforeEach( () => {
		// Create the block with the minimum attributes.
		block = createBlock( name );

		// Reset the reused variables.
		serializedBlock = '';
	} );

	it( 'should render with content', () => {
		block.attributes.title = 'Food item title';
		block.attributes.description = 'Food item description';
		block.attributes.price = '$1.00';
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'Food item title' );
		expect( serializedBlock ).toContain( 'Food item description' );
		expect( serializedBlock ).toContain( '$1.00' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should not render without content', () => {
		serializedBlock = serialize( createBlock( name ) );
		expect( serializedBlock ).toEqual( `<!-- wp:${ name } /-->` );
	} );
} );
