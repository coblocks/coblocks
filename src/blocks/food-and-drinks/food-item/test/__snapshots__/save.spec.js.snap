// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coblocks/food-item should render with content 1`] = `
"<!-- wp:coblocks/food-item -->
<div class=\\"wp-block-coblocks-food-item\\" itemscope itemtype=\\"http://schema.org/MenuItem\\"><div class=\\"wp-block-coblocks-food-item__content\\"><div class=\\"wp-block-coblocks-food-item__heading-wrapper\\"><h4 class=\\"wp-block-coblocks-food-item__heading\\" itemprop=\\"name\\">Food item title</h4></div><p class=\\"wp-block-coblocks-food-item__description\\" itemprop=\\"description\\">Food item description</p><p class=\\"wp-block-coblocks-food-item__price\\" itemprop=\\"offers\\" itemscope itemtype=\\"http://schema.org/Offer\\"><span itemprop=\\"price\\">$1.00</span></p></div></div>
<!-- /wp:coblocks/food-item -->"
`;
